### ISSUE TITLE
PNS - Possibility of Blocking the Start of Subsequent Auctions

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2023-09-nounsbuilder-judging#204]
PNS

medium

# Possibility of Blocking the Start of Subsequent Auctions

## Summary

A single bidder can delay the settle of an auction by repeatedly bidding on their own offer, thereby preventing the start of subsequent auctions.

## Vulnerability Detail

The [`_createBid()`](https://github.com/sherlock-audit/2023-09-nounsbuilder/blob/main/nouns-protocol/src/auction/Auction.sol#L158) function allows to postpone the end of the auction by bidding at the last minute (timeBuffer). 
Additionally, there is no mechanism preventing a bidder from outbidding their own offer. 
This allows a bidder to repeatedly outbid themselves, delaying the end of the auction. This, in turn, blocks the start of subsequent auctions, which is only possible after resolving the previous one and can only occur for auctions whose time has elapsed. 

The only protection against such behavior is the minimum next bid mechanism, which, depending on the settings, may be as low as 1%. Given a threshold setting of 1% and the fact that funds from the previous bid are immediately returned, a dishonest bidder can freeze the auction system for an extended period.

## Impact

Blocking the start of subsequent auctions.

#### Extending the Auction End
```solidity
File: src/auction/Auction.sol
189:             // Compute whether the time remaining is less than the buffer
190:             extend = (auction.endTime - block.timestamp) < settings.timeBuffer;
191: 
192:             // If the auction should be extended
193:             if (extend) {
194:                 // Update the end time with the additional time buffer
195:                 auction.endTime = uint40(block.timestamp + settings.timeBuffer);
196:             }
```

#### Starting a New Auction Only After Resolving the Previous One
```solidity
File: src/auction/Auction.sol
238:     function settleCurrentAndCreateNewAuction() external nonReentrant whenNotPaused {
239:         _settleAuction();
240:         _createAuction();
241:     }
```

## Tool used

Manual Review

## Recommendation 

Implement a restriction to prevent the possibility of outbidding one's own offer. This could be achieved through auction settings or a constant restriction.

#### Example Restriction to Prevent Self-Bidding
```solidity
File: src/auction/Auction.sol
172:         // Cache the address of the highest bidder
173:         address lastHighestBidder = auction.highestBidder;
174:         if(msg.sender == lastHighestBidder){
175:             revert INVALID_BIDDER();
176:         }
```
